snmp/datadog_checks/snmp/types.py

```diff
@@ -0,0 +1,23 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+"""
+Type declarations, for type checking purposes only.
+"""
+from typing import Literal, Set, TypedDict, Union
+
+ForceableMetricType = Literal['gauge', 'percent']
+MetricDefinition = TypedDict(
+    'MetricDefinition', {'type': Literal['gauge', 'rate', 'counter', 'monotonic_count'], 'value': float}
+)
+
+# SNMP value types that we support.
+# NOTE: these literals purposefully do NOT follow the names of PySNMP value classes.
+# This is because they should be completely decoupled from PySNMP classes.
+# Mapping from PySNMP classes to these literals is done elsewhere.
+SNMPCounterType = Literal['counter32', 'counter64', 'zero-based-counter64']
+SNMPGaugeType = Literal['gauge32', 'unsigned32', 'counter-based-gauge64', 'integer', 'integer32']
+SNMPType = Union[SNMPCounterType, SNMPGaugeType, Literal['opaque']]
+
+SNMP_COUNTERS = {'counter32', 'counter64', 'zero-based-counter64'}  # type: Set[SNMPCounterType]
+SNMP_GAUGES = {'gauge32', 'unsigned32', 'counter-based-gauge64', 'integer', 'integer32'}  # type: Set[SNMPGaugeType]
```

Comment:
I really don't see the point. Those are not pysnmp types, those are RFC documented SNMP types. I don't see what this translation layer brings.

Reply:
Yes, I agree. I'll remove these types and simplify the code.

Reply:
Addressed in c89f811

Reply:
Thanks that looks much better.